### DISCUSSION TITLE
vi.mock gives error "is not a function"

### DISCUSSION THREAD

**Discussion by steven-twerdochlib** [vitest-dev/vitest#5952] [Q&A]
Hi,
I'm converting a project from Jest to Vitest and my Jest has the following code:
`jest.mock('@/store', () => ({
  dispatch: jest.fn()
}));`

I've tried converting this to vitest like this:
`vi.mock('@/store', async () => {
  const mod = await vi.importActual('@/store');
  return {
    ...mod,
    dispatch: () => vi.fn(),
    default: mod
  };
});`

But this gives the error:
"TypeError: default.dispatch is not a function"

Any suggestions would be greatly appreciated

**Comment by sheremet-va** [comment]
**default**.dispatch should give you a hint that the `dispatch` function should be defined on default: https://vitest.dev/guide/migration.html#module-mocks

```ts
vi.mock('@/store', () => {
  const mod = await vi.importActual('@/store');
  const dispatch = vi.fn()
  return {
    default: {
      ...mod.default,
      dispatch,
    }
  }
})
```
